app/helpers/samples_helper.rb

```diff
@@ -142,14 +145,25 @@ def parsed_samples_for_s3_path(s3_path, project_id, host_genome_id)
     default_attributes = { project_id: project_id.to_i,
                            host_genome_id: host_genome_id,
                            status: 'created', }
-    s3_path.chomp!('/')
-    s3_bucket = 's3://' + s3_path.sub('s3://', '').split('/')[0]
-    s3_output, _stderr, status = Open3.capture3("aws", "s3", "ls", "--recursive", "#{s3_path}/")
-    return unless status.exitstatus.zero?
-    s3_output.chomp!
-    entries = s3_output.split("\n").reject { |line| line.include? "Undetermined" }
+
+    parsed_uri = URI.parse(s3_path)
+    return unless parsed_uri.scheme == "s3"
```

Comment:
Can we use if like below? `if parsed_uri.scheme != "s3"`

Reply:
I don't like unless either, but I'm following conventions from here: https://github.com/rubocop-hq/ruby-style-guide#unless-for-negatives

Reply:
ok, if it must...

Reply:
I was trying to follow the unless mantra, but in this case I think it makes sense